Backpack/src/main/java/net/skyscanner/backpack/text/BpkTextField.kt

```diff
@@ -68,6 +68,12 @@ open class BpkTextField @JvmOverloads constructor(
       setCompoundDrawablesRelativeWithIntrinsicBounds(iconStart, null, iconEnd, null)
     }
 
+  var hasError: Boolean = false
```

Comment:
this is to add an error border in case of an error mode

Reply:
This is potentially dangerous situation as we now has 2 source of truth for error (border display flag):
* drawable state
* boolean flag

I wonder if we can use the drawable state as backing field here, e.g. `get() = android.R.attr.state_error in drawableState`

Reply:
Unfortunately this doesn't work as to add it to the drawable state we need to refresh the drawable state which then needs to know if we need to add it to the drawable state, so we end up in a stackoverflow error - we can't add it directly

Reply:
What if instead of using field there we'll check it in array itself?

Reply:
It won't be in the array without it being added in the `onCreateDrawableState` method